### ISSUE TITLE
Copying text from commlong loses some characters when it starts a wrapped line

### ISSUE THREAD

**Issue by AardCrowley** [fiendish/aardwolfclientpackage#292]
When you copy text from the commlog (with or without colors), the copy drops text. This seems to happen when copying lines that wrap to the next line. The first letter of the wrapped line seems to get dropped occasionally. The best way to reproduce this is to fill a tab with lines of text to where it wraps around (could be easier to shrink the commlog some so you don't have to type so much). After several lines, highlight the text, copy with or without colors, and paste somewhere. Even in the main window you'll find missing letters, but I found it's helpful to paste in Word for spellcheck to see the missing letters.

**Comment by fiendish** [comment]
please replace your lua/text_rect.lua with https://raw.githubusercontent.com/fiendish/aardwolfclientpackage/844db73b945e6551e52c8f4eba4a3eba6bdae4a5/MUSHclient/lua/text_rect.lua and try again

**Comment by AardCrowley** [comment]
Replaced text_rect.lua, and it seems to work, though awkwardly it adds "@w" before the start of each line, for example:
`@w[30 May 14:29:25] @x011(@x015Group@x011) @x110@MCrowley@x110: 'When you copy text from the commlog (with or without colors), the copy drops text. This seems to happen when copying lines that wrap to the @w@x110next line. The first letter of the wrapped line seems to get dropped occasionally. The best way to reproduce this is to fill a tab with lines of text to where it wraps around @w@x110(could be easier to shrink the commlog some so you don't have to type so much). After several lines, highlight the text, copy with or without colors, and paste somewhere. Even @w@x110in the main window you'll find missing letters, but I found it's helpful to paste in Word for spellcheck to see the missing letters.'@w`

For reference,

![image](https://user-images.githubusercontent.com/12467582/83336579-abccdf00-a282-11ea-8256-34c271f47c7f.png)

**Comment by fiendish** [comment]
Ok please try with these
https://raw.githubusercontent.com/fiendish/aardwolfclientpackage/76d9126a673ae0ca9308f68a1ced988d35f63c64/MUSHclient/worlds/plugins/aardwolf_colors.lua
https://raw.githubusercontent.com/fiendish/aardwolfclientpackage/76d9126a673ae0ca9308f68a1ced988d35f63c64/MUSHclient/lua/text_rect.lua

**Comment by AardCrowley** [comment]
Replaced, and it still produces the same output of `@w@x123` at line breaks.

**Comment by fiendish** [comment]
I'm not seeing it. 
<img width="769" alt="Screen Shot 2020-05-31 at 5 29 15 PM" src="https://user-images.githubusercontent.com/201996/83363615-bc578500-a368-11ea-9798-0b9d46c55ca7.png">
<img width="1607" alt="Screen Shot 2020-05-31 at 5 29 21 PM" src="https://user-images.githubusercontent.com/201996/83363616-bcf01b80-a368-11ea-881b-3bf570319ce9.png">

Can you give me steps to reproduce? (and did you get both files? or maybe just clone the development branch)

**Comment by AardCrowley** [comment]
Well, that's a bit of a head scratcher, then. I used the same output spam on group channel that you used, and I come back with:

![image](https://user-images.githubusercontent.com/12467582/83363823-861b0500-a36a-11ea-8269-845e0d4a4d43.png)

I don't really have steps that can reproduce this, either. I've tried shrinking and expanding the comm log window to see if it has any effect, and it doesn't. Ultimately, the additional `@w` before the xterm color code does not hurt anything, If I can find a reliable method of reproducing it, I'll reraise the issue, but I imagine at this point, it's not very helpful if you have no means of reproducing it yourself.

**Comment by fiendish** [comment]
Maybe a problem with the file urls? Try https://github.com/fiendish/aardwolfclientpackage/archive/development.zip

**Comment by AardCrowley** [comment]
I don't think it was a problem with the file urls; I did a compare before replacing each one to see what was changed.

No, I think it a problem with the end user in this case. See, I was running MUSHclient when I copied the files over. They're .lua files, so I figured copying them over would suffice, then I could close out MUSH and restart. I don't think the Lua file save. This time, I closed out MUSH, copied the files over from the zip, and opened MUSH up again. Ran the test, and I get the same results you do now.

Next time, I'll have to remember to close out MUSH first before copying things over. Thanks for fixing this!
